codalab/model/bundle_model.py

```diff
@@ -539,113 +543,114 @@ def make_condition(key, field, value):
                     )
                 )
-                clause = or_(*clause)
+                where_clause = or_(*where_clause)
             elif key == '':  # Match any field
-                clause = []
-                clause.append(cl_bundle.c.uuid.like('%' + value + '%'))
-                clause.append(cl_bundle.c.command.like('%' + value + '%'))
-                clause.append(
-                    cl_bundle.c.uuid.in_(
-                        alias(
-                            select([cl_bundle_metadata.c.bundle_uuid]).where(
-                                cl_bundle_metadata.c.metadata_value.like('%' + value + '%')
-                            )
-                        )
-                    )
+                aliased_bundle_metadata = aliased(cl_bundle_metadata)
+                add_join(
+                    aliased_bundle_metadata,
+                    cl_bundle.c.uuid == aliased_bundle_metadata.c.bundle_uuid,
                 )
-                clause = or_(*clause)
+
+                where_clause = []
+                where_clause.append(cl_bundle.c.uuid.like('%' + value + '%'))
+                where_clause.append(cl_bundle.c.command.like('%' + value + '%'))
+                where_clause.append(
+                    aliased_bundle_metadata.c.metadata_value.like('%' + value + '%')
+                )
+                where_clause = or_(*where_clause)
             # Otherwise, assume metadata.
             else:
-                condition = make_condition(key, cl_bundle_metadata.c.metadata_value, value)
+                aliased_bundle_metadata = aliased(cl_bundle_metadata)
+                condition = make_condition(key, aliased_bundle_metadata.c.metadata_value, value)
                 if condition is None:  # top-level
-                    clause = and_(
-                        cl_bundle.c.uuid == cl_bundle_metadata.c.bundle_uuid,
-                        cl_bundle_metadata.c.metadata_key == key,
+                    where_clause = and_(
+                        cl_bundle.c.uuid == aliased_bundle_metadata.c.bundle_uuid,
+                        aliased_bundle_metadata.c.metadata_key == key,
                     )
                 else:  # embedded
-                    clause = cl_bundle.c.uuid.in_(
-                        select([cl_bundle_metadata.c.bundle_uuid]).where(
-                            and_(cl_bundle_metadata.c.metadata_key == key, condition)
-                        )
+                    add_join(
+                        aliased_bundle_metadata,
+                        cl_bundle.c.uuid == aliased_bundle_metadata.c.bundle_uuid,
                     )
 
-            if clause is not None:
-                clauses.append(clause)
+                    where_clause = and_(aliased_bundle_metadata.c.metadata_key == key, condition)
 
-        clause = and_(*clauses)
+            if where_clause is not None:
+                where_clauses.append(where_clause)
+
+        where_clause = and_(*where_clauses)
```

Comment:
Might be confusing to reuse `where_clause` for both the top-level clause and the conjuncts.  One might consider creating helper function(s) to break up this big function.

Reply:
Done.